clients/admin-ui/src/features/config-wizard/AuthenticateOktaForm.tsx

```diff
@@ -0,0 +1,214 @@
+import { useAppDispatch, useAppSelector } from "~/app/hooks";
+import DocsLink from "~/features/common/DocsLink";
+import { CustomTextInput } from "~/features/common/form/inputs";
+import {
+  isErrorResult,
+  ParsedError,
+  parseError,
+} from "~/features/common/helpers";
+import { useAlert } from "~/features/common/hooks";
+import {
+  Dataset,
+  GenerateResponse,
+  GenerateTypes,
+  System,
+  ValidTargets,
+} from "~/types/api";
+import { RTKErrorResult } from "~/types/errors";
+
+import {
+  changeStep,
+  selectOrganizationFidesKey,
+  setSystemsForReview,
+} from "./config-wizard.slice";
+import { DOCS_URL_OKTA_TOKEN } from "./constants";
+import { useGenerateMutation } from "./scanner.slice";
+import ScannerError from "./ScannerError";
+import ScannerLoading from "./ScannerLoading";
+
+const isSystem = (sd: System | Dataset): sd is System => "system_type" in sd;
+
+const initialValues = {
+  orgUrl: "",
+  token: "",
+};
+
+type FormValues = typeof initialValues;
+
+const ValidationSchema = Yup.object().shape({
+  orgUrl: Yup.string()
+    .required()
+    .trim()
+    .matches(/^[^\s]+$/, "Cannot contain spaces")
+    .label("URL"),
+  token: Yup.string()
+    .required()
+    .trim()
+    .matches(/^[^\s]+$/, "Cannot contain spaces")
+    .label("Token"),
+});
+
+const AuthenticateOktaForm = () => {
+  const organizationKey = useAppSelector(selectOrganizationFidesKey);
+  const dispatch = useAppDispatch();
+  const { successAlert } = useAlert();
+
+  const [scannerError, setScannerError] = useState<ParsedError>();
+
+  const handleResults = (results: GenerateResponse["generate_results"]) => {
+    const systems: System[] = (results ?? []).filter(isSystem);
+    dispatch(setSystemsForReview(systems));
+    dispatch(changeStep());
+    successAlert(
+      `Your scan was successfully completed, with ${systems.length} new systems detected!`,
+      `Scan Successfully Completed`,
+      { isClosable: true }
+    );
+  };
+  const handleError = (error: RTKErrorResult["error"]) => {
+    const parsedError = parseError(error, {
+      status: 500,
+      message: "Our system encountered a problem while connecting to Okta.",
+    });
+    setScannerError(parsedError);
+  };
+  const handleCancel = () => {
+    dispatch(changeStep(2));
+  };
+
+  const [generate, { isLoading }] = useGenerateMutation();
+
+  const handleSubmit = async (values: FormValues) => {
+    setScannerError(undefined);
+
+    const result = await generate({
+      organization_key: organizationKey,
+      generate: {
+        config: values,
+        target: ValidTargets.OKTA,
+        type: GenerateTypes.SYSTEMS,
+      },
+    });
+
+    if (isErrorResult(result)) {
+      handleError(result.error);
+    } else {
+      handleResults(result.data.generate_results);
+    }
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={ValidationSchema}
+      onSubmit={handleSubmit}
+    >
+      {({ isValid, isSubmitting, dirty }) => (
```

Comment:
You can use something like this `{(props: FormikProps<typeof initialValues>) => (`. Then just reference `props.isSubmitting` for example. This will give you a strongly data type and intellisense.